hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala

```diff
@@ -75,6 +75,9 @@ abstract class HoodieBaseHadoopFsRelationFactory(val sqlContext: SQLContext,
   protected lazy val basePath: StoragePath = metaClient.getBasePath
   protected lazy val partitionColumns: Array[String] = tableConfig.getPartitionFields.orElse(Array.empty)
 
+  //placeholder until pr to persist field types lands
+  protected lazy val partitionColumnsToRead: Array[String] = if (shouldExtractPartitionValuesFromPartitionPath) Array.empty else tableConfig.getPartitionFields.orElse(Array.empty)
```

Comment:
I think we should deprecate `hoodie.datasource.read.extract.partition.values.from.path`.  For timestamp-typed partition column, the query should directly read values from the data files.

Reply:
https://issues.apache.org/jira/browse/HUDI-8100